packages/cache-memcached/src/services/memcached-cache.ts

```diff
@@ -0,0 +1,99 @@
+import Memcached from "memcached"
+
+import { ICacheService } from "@medusajs/types"
+
+import { MemcachedCacheModuleOptions } from "../types"
+
+const DEFAULT_NAMESPACE = "medusa"
+const DEFAULT_CACHE_TIME = 30 // 30 seconds
+
+type InjectedDependencies = {
+  cacheMemcachedConnection: Memcached
+}
+
+class MemcachedCacheService implements ICacheService {
+  protected readonly TTL: number
+  private readonly namespace: string
+  protected readonly memcached: Memcached
+
+  constructor(
+    { cacheMemcachedConnection }: InjectedDependencies,
+    options: MemcachedCacheModuleOptions
+  ) {
+    this.memcached = cacheMemcachedConnection
+    this.TTL = options.ttl || DEFAULT_CACHE_TIME
```

Comment:
todo: use ?? instead, otherwise if a user want to set it to 0 it will fallback to the default